### DISCUSSION TITLE
[Bug] why character number is not found in the vocabulary?

### DISCUSSION THREAD

**Discussion by tangzhimiao** [coqui-ai/TTS#2450] [General Q&A]
### Describe the bug

i use fastspeech2 model train my data voice model, when i synthesize wav file and play , listen the wav file prosody not right
 

### To Reproduce

synthesize command:
from TTS.api import TTS
tts = TTS(model_path="/../../best_model_596750.pth", config_path="/../../config.json", progress_bar=False, gpu=False)
tts.tts_to_file(text="我是张先生", file_path="./output.wav")

### Expected behavior

> Using model: fastspeech2
 > Setting up Audio Processor...
 | > sample_rate:22050
 | > resample:False
 | > num_mels:80
 | > log_func:np.log
 | > min_level_db:-100
 | > frame_shift_ms:None
 | > frame_length_ms:None
 | > ref_level_db:20
 | > fft_size:1024
 | > power:1.5
 | > preemphasis:0.0
 | > griffin_lim_iters:60
 | > signal_norm:False
 | > symmetric_norm:True
 | > mel_fmin:0
 | > mel_fmax:8000.0
 | > pitch_fmin:1.0
 | > pitch_fmax:640.0
 | > spec_gain:1.0
 | > stft_pad_mode:reflect
 | > max_norm:4.0
 | > clip_norm:True
 | > do_trim_silence:True
 | > trim_db:60
 | > do_sound_norm:False
 | > do_amp_to_db_linear:True
 | > do_amp_to_db_mel:True
 | > do_rms_norm:False
 | > db_level:None
 | > stats_path:None
 | > base:2.718281828459045
 | > hop_length:256
 | > win_length:1024
 > Text splitted to sentences.

### Logs

```shell
['我是张先生']
text 我是张先生
Building prefix dict from the default dictionary ...
DEBUG:jieba:Building prefix dict from the default dictionary ...
Loading model from cache /tmp/jieba.cache
DEBUG:jieba:Loading model from cache /tmp/jieba.cache
Loading model cost 0.636 seconds.
DEBUG:jieba:Loading model cost 0.636 seconds.
Prefix dict has been built successfully.
DEBUG:jieba:Prefix dict has been built successfully.
result ['w', 'o', '3', ' ', 'ʂ', 'ʏ', '4', ' ', 'd', 'ʒ', 'ɑ', 'ŋ', '1', ' ', 'ɕ', 'i', 'ɛ', 'n', '1', 'ʂ', 'ɵ', 'ŋ', '1']
seperator.join wo3 ʂʏ4 dʒɑŋ1 ɕiɛn1ʂɵŋ1
idx 25 w
idx 17 o
wo3 ʂʏ4 dʒɑŋ1 ɕiɛn1ʂɵŋ1
 [!] Character '3' not found in the vocabulary. Discarding it.
idx 130  
idx 85 ʂ
idx 95 ʏ
wo3 ʂʏ4 dʒɑŋ1 ɕiɛn1ʂɵŋ1
 [!] Character '4' not found in the vocabulary. Discarding it.
idx 130  
idx 7 d
idx 98 ʒ
idx 41 ɑ
idx 34 ŋ
wo3 ʂʏ4 dʒɑŋ1 ɕiɛn1ʂɵŋ1
 [!] Character '1' not found in the vocabulary. Discarding it.
idx 130  
idx 45 ɕ
idx 11 i
idx 51 ɛ
idx 16 n
idx 85 ʂ
idx 75 ɵ
idx 34 ŋ
text_inputs [ 25  17 130  85  95 130   7  98  41  34 130  45  11  51  16  85  75  34]
 > Processing time: 1.5648632049560547
 > Real-time factor: 0.6981473306345298
```

### Environment

```shell
pytorch 1.11
python 3.7
TTS 0.11.1
gpu tesla
CUDA Version 11.0
```

### Additional context

https://user-images.githubusercontent.com/33514716/224881445-91d76973-3943-4825-b31a-75ae54c3d14b.mp4

**Comment by Edresson** [comment]
Hi @tangzhimiao,

As default :frog:  TTS does not include numbers in the vocabulary. You can check here the default vocabulary https://github.com/coqui-ai/TTS/blob/dev/TTS/tts/utils/text/characters.py#L18-L35. You can create a custom vocabulary that includes the numbers if you wish. [Here](https://github.com/coqui-ai/TTS/blob/3c15f0619a3c87aca6ed99c614d930575ecb7ee1/recipes/vctk/yourtts/train_yourtts.py#L173) is one example of how to do it.  Another way around it can be the use of one phonemizer that already deals with numbers expansion (like 
espeak-ng).
